src/PowerShellEditorServices/Language/FindDeclarationVisitor.cs

```diff
@@ -76,19 +78,68 @@ public override AstVisitAction VisitFunctionDefinition(FunctionDefinitionAst fun
         /// or a decision to continue if it wasn't found</returns>
         public override AstVisitAction VisitAssignmentStatement(AssignmentStatementAst assignmentStatementAst)
         {
-            var variableExprAst = assignmentStatementAst.Left as VariableExpressionAst;
-            if (variableExprAst == null ||
-                variableName == null ||
-                !variableExprAst.VariablePath.UserPath.Equals(
+            if (variableName == null)
+            {
+                return AstVisitAction.Continue;
+            }
+
+            // The AssignmentStatementAst could contain either of the following Ast types:
+            // VariableExpressionAst, ArrayLiteralAst, ConvertExpressionAst, AttributedExpressionAst
+            // We might need to recurse down the tree to find the VariableExpressionAst we're looking for
+            List<VariableExpressionAst> asts = FindMatchingAsts(assignmentStatementAst.Left, variableName);
+            if (asts.Count > 0)
+            {
+                FoundDeclaration = new SymbolReference(SymbolType.Variable, asts.FirstOrDefault().Extent);
```

Comment:
Will this only check the first variable when the lhs is an array literal?  If so we should enumerate until we find a match or the list is completed.

Reply:
May be worth looking at this method (in typical fashion, it's private, but still may be helpful):
https://github.com/PowerShell/PowerShell/blob/a3786158ca51cd65388743f900b69ec9e253c3d9/src/System.Management.Automation/engine/parser/ast.cs#L7367-L7379

Reply:
@SeeminglyScience that will grab the first in an array literal, I guess. But we probably want the LAST hit from an array literal.

```
$a, [string] $b = "asdf", 5, $b = 26, 5
$b
```

FindMatchingAsts for `$b` will give the 2 hits in a list...
 
we want `$b` to jump to `$b = 26` not `[string] $b = "asdf"` so I'll have it grab the last instead of the first - good catch.


@rjmholt looks like they went iterative. I like that, but it's a little extra code when we could just continue to recurse (since we're already doing it). Thanks for the reference though :) let me know if you feel strongly otherwise

Reply:
I prefer the clarity of recursion actually. But thought it was worth noting that an implementation already exists